spec/statement.md

```diff
@@ -0,0 +1,166 @@
+
+## Table of Contents
+
+* [Definitions](#definitions)
+  * [Current indentation level](#current-indentation-level)
+  * [Document indentation character](#document-indentation-character)
+* [Syntax](#syntax)
+  * [ScssStatements](#scssstatements)
+  * [IndentedStatements](#indentedstatements)
+  * [Statements](#statements)
+  * [Block](#block)
+  * [Comments](#comments)
+    * [LoudComment](#loudcomment)
+    * [SilentComment](#silentcomment)
+    * [WhitespaceComment](#whitespacecomment)
+  * [Whitespace](#whitespace)
+  * [Indentation](#indentation)
+
+## Definitions
+
+### Current indentation level
+
+The `current indentation level` is the count of the [`documentation indentation
+character`] between a new line and any non-whitespace characters of the last
+consumed statement. Lines that only contain whitespace or do not start a
+statement do not impact the `current indentation level`.
+
+A document parsed in the indented syntax may not begin with whitespace, and the
+initial `current indentation level` for any document is `0`.
+
+> Changes in the indentation level are used by the indented syntax to start and
+> end blocks of statements.
+
+[`documentation indentation character`]: #document-indentation-character
+
+### Document indentation character
+
+The `document indentation character` is defined as the first tab or space
+character used as an [`IndentMore`] production in a document.
+
+[`IndentMore`]: #indentation
+
+> The `document indentation character` is the character used for calculating the
+> [`current indentation level`]. In the indented syntax, no character other than
+> the `document indentation character` may be used for indentation.
+
+[`current indentation level`]: #current-indentation-level
+
+## Syntax
+
+### ScssStatements
+
+<x><pre>
+**ScssStatements**      ::= (Statement ';'?¹)* Statement?
+</pre></x>
+
+1: This production is mandatory unless the previous `Statement` is a
+`LoudComment`, `SilentComment`, or ends in a `Block`.
+
+If a `WhitespaceComment` would be ambiguous with a `Statement` in the
+`ScssStatements` rule, parse it preferentially as a `Statement`.
+
+### IndentedStatements
+
+<x><pre>
+**IndentedStatements**  ::= (Statement [IndentSame])* Statement
+</pre></x>
+
+[IndentSame]: #indentation
+
+### Statements
+
+<x><pre>
+**Statements**          ::= ([ScssStatements] | [IndentedStatements])¹
+</pre></x>
+
+[ScssStatements]: #scssstatements
+[IndentedStatements]: #indentedstatements
+
+1. Only the production for the current syntax is valid.
+
+### Block
+
+<x><pre>
+**ScssBlock**      ::= '{' [ScssStatements] '}'
+**IndentedBlock**  ::= [IndentMore] [IndentedStatements]
+**Block**          ::= (ScssBlock | IndentedBlock)¹
+</pre></x>
+
+[IndentMore]: #indentation
+
+1: Only the production for the current syntax is valid.
+
+### Comments
+
+#### LoudComment
+
+<x><pre>
+**ScssLoudComment**          ::= '/*' '!'? (.* | Interpolation)*'*/'
+**IndentedLoudComment**      ::= '/*' '!'? (.* | Interpolation)*
```

Comment:
Similarly, this `.*` needs to indicate that it can't contain newlines. (Similarly below.)

Also: statement-level comments in the indented syntax allow nested content. (It's weird, I know.)